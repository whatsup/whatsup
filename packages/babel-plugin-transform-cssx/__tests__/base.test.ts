/**
 * @jest-environment jest-environment-node-single-context
 */

import { transform } from './__config__/transform'

describe('transform types', function () {
    it('Should transform css modules', function () {
        expect(transform('./modules.tsx')).toMatchSnapshot()
    })

    it('Should transform cssx', function () {
        expect(transform('./cssx.tsx')).toMatchSnapshot()
    })

    it('Should transform mixed', function () {
        expect(transform('./mixed.tsx')).toMatchSnapshot()
    })

    it('Should transform simle css', function () {
        expect(transform('./simple_css.tsx')).toMatchSnapshot()
    })
})
