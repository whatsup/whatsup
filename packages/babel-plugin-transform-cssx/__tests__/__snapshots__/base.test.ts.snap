// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform types Should transform css modules 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

require(\\"./component.scss\\");

var _cssx = require(\\"@whatsup/cssx\\");

const styles_UniqueId = {
  \\"col_2_1\\": \\"component__col_2_1-HwdLD\\",
  \\"col_1_3\\": \\"component__col_1_3-32aDq\\",
  \\"button\\": \\"component__button-IICp7\\",
  \\"active\\": \\"component__active-2vXAv\\",
  \\"red\\": \\"component__red-yqInI\\"
};
const styles = styles_UniqueId;

var _default = () => {
  styles;
};

exports.default = _default;"
`;

exports[`transform types Should transform cssx 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

require(\\"./component.scss\\");

var _cssx = require(\\"@whatsup/cssx\\");

const styles_UniqueId = {
  \\"col_2_1\\": \\"component__col_2_1-HwdLD\\",
  \\"col_1_3\\": \\"component__col_1_3-32aDq\\",
  \\"button\\": \\"component__button-IICp7\\",
  \\"active\\": \\"component__active-2vXAv\\",
  \\"red\\": \\"component__red-yqInI\\"
};
const Div = (0, _cssx.createComponent)(\\"div\\", styles_UniqueId);
const Span = (0, _cssx.createComponent)(\\"span\\", styles_UniqueId);

var _default = () => {
  Div;
  Span;
};

exports.default = _default;"
`;

exports[`transform types Should transform mixed 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

require(\\"./component.scss\\");

var _cssx = require(\\"@whatsup/cssx\\");

const styles_UniqueId = {
  \\"col_2_1\\": \\"component__col_2_1-HwdLD\\",
  \\"col_1_3\\": \\"component__col_1_3-32aDq\\",
  \\"button\\": \\"component__button-IICp7\\",
  \\"active\\": \\"component__active-2vXAv\\",
  \\"red\\": \\"component__red-yqInI\\"
};
const styles = styles_UniqueId;
const Div = (0, _cssx.createComponent)(\\"div\\", styles_UniqueId);
const Span = (0, _cssx.createComponent)(\\"span\\", styles_UniqueId);

var _default = () => {
  styles;
  Div;
  Span;
};

exports.default = _default;"
`;

exports[`transform types Should transform simle css 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

require(\\"./component.css\\");

var _cssx = require(\\"@whatsup/cssx\\");

const styles_UniqueId = {
  \\"button\\": \\"component__button-2ho2o\\",
  \\"active\\": \\"component__active-2gcOb\\",
  \\"red\\": \\"component__red-G7ajg\\"
};
const styles = styles_UniqueId;
const Div = (0, _cssx.createComponent)(\\"div\\", styles_UniqueId);
const Span = (0, _cssx.createComponent)(\\"span\\", styles_UniqueId);

var _default = () => {
  styles;
  Div;
  Span;
};

exports.default = _default;"
`;
